io_scene_vrm/importer/vrm_animation_importer.py

```diff
@@ -0,0 +1,784 @@
+    buffer_dicts: list[Json],
+    buffer0_bytes: bytes,
+) -> Optional[list[Quaternion]]:
+    if accessor_dict.get("type") != "VEC4":
+        return None
+    buffer_bytes = read_accessor_as_bytes(
+        accessor_dict, buffer_view_dicts, buffer_dicts, buffer0_bytes
+    )
+    if buffer_bytes is None:
+        return None
+    unpacked_values = unpack_vector_accessor(accessor_dict, buffer_bytes)
+    if not unpacked_values:
+        return None
+    return [Quaternion((w, x, -z, y)).normalized() for x, y, z, w in unpacked_values]
+
+
+def work_in_progress(
+    context: bpy.types.Context, path: Path, armature: bpy.types.Object
+) -> set[str]:
+    humanoid = armature.data.vrm_addon_extension.vrm1.humanoid
+    if not humanoid.human_bones.all_required_bones_are_assigned():
+        return {"CANCELLED"}
+
+    saved_pose_position = armature.data.pose_position
+    vrm1 = armature.data.vrm_addon_extension.vrm1
+
+    # TODO: 現状restがTポーズの時しか動作しない
+    # TODO: 自動でTポーズを作成する
+    # TODO: Tポーズ取得処理、共通化
+    try:
+        if context.view_layer.objects.active is not None:
+            bpy.ops.object.mode_set(mode="OBJECT")
+        bpy.ops.object.select_all(action="DESELECT")
+        context.view_layer.objects.active = armature
+        bpy.ops.object.mode_set(mode="POSE")
+
+        armature.data.pose_position = "POSE"
+
+        t_pose_action = vrm1.humanoid.pose_library
+        t_pose_pose_marker_name = vrm1.humanoid.pose_marker_name
+        pose_marker_frame = 0
+        if t_pose_pose_marker_name:
+            for search_pose_marker in t_pose_action.pose_markers.values():
+                if search_pose_marker.name == t_pose_pose_marker_name:
+                    pose_marker_frame = search_pose_marker.frame
+                    break
+
+        bpy.context.view_layer.update()
+
+        if t_pose_action:
+            armature.pose.apply_pose_from_action(
+                t_pose_action, evaluation_time=pose_marker_frame
+            )
+        else:
+            for bone in armature.pose.bones:
+                bone.location = (0, 0, 0)
+                bone.scale = (1, 1, 1)
+                if bone.rotation_mode != "QUATERNION":
+                    bone.rotation_mode = "QUATERNION"
+                bone.rotation_quaternion = Quaternion()
+
+        bpy.context.view_layer.update()
+
+        return work_in_progress_2(context, path, armature)
+    finally:
+        # TODO: リストア処理、共通化
+        if armature.data.pose_position != "POSE":
+            armature.data.pose_position = "POSE"
+        if context.view_layer.objects.active is not None:
+            bpy.ops.object.mode_set(mode="OBJECT")
+        bpy.ops.object.select_all(action="DESELECT")
+        context.view_layer.objects.active = armature
+        bpy.context.view_layer.update()
+
+        if saved_pose_position:
+            armature.data.pose_position = saved_pose_position
+        bpy.ops.object.mode_set(mode="OBJECT")
+
+
+def work_in_progress_2(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this function is 53. By comparison, 99% of the functions in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the function has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 428-439 into a separate function.